### ISSUE TITLE
Warn if waiting too long for server connection

### ISSUE THREAD

**Issue by blairlyons** [allen-cell-animated/simularium-viewer#169]
## Use Case
While we're working to set up autoscaling on the backend, if too many concurrent users use the website, they could overload the server. It would be nice if a client could warn them that the server is under high load rather than the blank viewer spinning forever.

## Acceptance Criteria
An app installing the viewer gets a warning if the viewer has been waiting on a connection request for more than 4(?) sec

## Details
Eventually the server should send a warning if it's overloaded, but until we get to that devops work, we can use a timeout to give the user some feedback.

Possible scenarios of issues with websockets connection:
**1. fail to establish connection**
2. connect, but becomes invalid
3. valid connection, but slow operation (lower priority since we're prioritizing posted examples which have been cached)

This handles the case when we fail to connect (1).

**Comment by schoinh** [comment]
@EricJIsaac I'm trying to figure out where exactly to start a timer. Which websocket request by the viewer would a server overload cause to hang? Is it [this one](https://github.com/allen-cell-animated/simularium-viewer/blob/a243588ee94f8c964b7b0683a46abd2007a176c6/src/simularium/RemoteSimulator.ts#L408)?

**Comment by schoinh** [comment]
Maybe we don't necessarily need to pinpoint the exact websocket request? I guess I could just start a timer at the beginning of [`SimulariumController.changeFile()`](https://github.com/allen-cell-animated/simularium-viewer/blob/a243588ee94f8c964b7b0683a46abd2007a176c6/src/controller/index.ts#L276) since that is what's called by simularium-website's [`loadNetworkedFile`](https://github.com/allen-cell-animated/simularium-website/blob/dfb768b913e18cb52bcc593d0fb53bbb960d48f1/src/state/trajectory/logics.ts#L137) logic 🤔 

Kinda tricky to build this out without being able to actually trigger the warning. Is there a good way to simulate a server overload or something similar? 

@meganrm @toloudis ideas welcome :)

**Comment by toloudis** [comment]
https://github.com/allen-cell-animated/simularium-viewer/blob/main/src/simularium/RemoteSimulator.ts#L267
There's a timeout in there too, but I'm not sure if it's affected under heavy load.

**Comment by toloudis** [comment]
And I imagine you could mock it to trigger the warning in the unit tests! there's already a mock "remote connection" in there

**Comment by EricJIsaac** [comment]
If working properly, [this](https://github.com/allen-cell-animated/simularium-viewer/blob/a243588ee94f8c964b7b0683a46abd2007a176c6/src/simularium/RemoteSimulator.ts#L293) should be called if the client is waiting too long for a server connection. After this the connection stays open.

One way to approach this task could be to propagate this promise rejection upward.
